src/mlpack/tests/one_hot_encoding_test.cpp

```diff
@@ -0,0 +1,211 @@
+  arma::Mat<int> input;
+  input = "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;";
+
+  matrix = "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;";
+
+  // Output matrix to save one-hot encoding results.
+  arma::Mat<int> output;
+  arma::Col<size_t> indices("1 3");
+  data::OneHotEncoding(input, indices, output);
+
+  REQUIRE(matrix.n_cols == output.n_cols);
+  REQUIRE(matrix.n_rows == output.n_rows);
+  for (size_t i = 0; i < output.n_elem; i++)
+    REQUIRE(matrix.at(i) == output.at(i));
+}
+
+TEST_CASE("OneHotEncodingVeryBigInputTest", "[OneHotEncodingTest]")
+{
+  arma::Mat<int> matrix;
+  arma::Mat<int> input;
+  input = "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "1 1 -1 -1 -1 -1 1 1;";
+
+  matrix = "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 1 -1 -1 -1 -1 1 1;";
+
+  // Output matrix to save one-hot encoding results.
+  arma::Mat<int> output;
+  arma::Col<size_t> indices("1 4 7 8");
+  data::OneHotEncoding(input, indices, output);
+
+  REQUIRE(matrix.n_cols == output.n_cols);
+  REQUIRE(matrix.n_rows == output.n_rows);
+  for (size_t i = 0; i < output.n_elem; i++)
+    REQUIRE(matrix.at(i) == output.at(i));
+}
+
+TEST_CASE("OneHotEncodingDatasetinfoTest", "[OneHotEncodingTest]")
+{
+  fstream f;
+  f.open("test.csv", fstream::out);
+  f << "1, 2, hello" << endl;
+  f << "3, 4, goodbye" << endl;
+  f << "5, 6, coffee" << endl;
+  f << "7, 8, confusion" << endl;
+  f << "9, 10, hello" << endl;
+  f << "11, 12, confusion" << endl;
+  f << "13, 14, confusion" << endl;
+  f.close();
+
+  // Load the test CSV.
+  arma::umat matrix;
+  DatasetInfo info;
+  data::Load("test.csv", matrix, info);
+  arma::umat output;
+  data::OneHotEncoding(matrix, output, info);
+  REQUIRE(output.n_cols == 7);
+  REQUIRE(output.n_rows == 6);
+  REQUIRE(info.Type(0) == Datatype::numeric);
+  REQUIRE(info.Type(1) == Datatype::numeric);
+  REQUIRE(info.Type(2) == Datatype::categorical);
+
+  REQUIRE(output(0, 0) == 1);
+  REQUIRE(output(1, 0) == 2);
+  REQUIRE(output(2, 0) == 1);
```

Comment:
I know it's tedious, but should we also check that `output(3, 0)` and `output(4, 0)` and `output(5, 0)` are 0 here?